azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py

```diff
@@ -62,6 +64,7 @@ def _run_op(self, op):
         elif isinstance(op, pipeline_ops_base.ConnectOperation):
             logger.info("{}({}): conneting".format(self.name, op.name))
 
+            @pipeline_thread.invoke_on_pipeline_thread_nowait
```

Comment:
I'm unclear as to why this has to specifically be invoked on the pipeline thread, when the outer function is clearly running on the pipeline thread (due to the assertion decorator)

Reply:
Good question.  This is invoked by paho after the outer function returns.  (side note: I'm revisiting this usage as part of my paho errors PR, so this will change).

---
In reply to: [305463891](https://github.com/Azure/azure-iot-sdk-python-preview/pull/133#discussion_r305463891) [](ancestors = 305463891)